custom/icds_reports/static/js/angular-services/template-provider.service.js

```diff
@@ -0,0 +1,12 @@
+window.angular.module('icdsApp').factory('templateProviderService', ['isMobile', function (isMobile) {
+    var url = hqImport('hqwebapp/js/initial_page_data').reverse;
+    return {
+        getTemplate: function (templateName) {
```

Comment:
helper function to dynamically swap in a template based on a context variable controlling whether it's the mobile version or not